### ISSUE TITLE
RuntimeError: failed opening file: Too many open files: iostream error

### ISSUE THREAD

**Issue by paulperry** [citp/BlockSci#67]
On an EC2 started Sunday 3/18/18 from the AMI you provide I get the following error after using it since it started.
```
cluster_data_directory = "/home/ubuntu/bitcoin"
chain = blocksci.Blockchain(cluster_data_directory)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-8-d798b56482d9> in <module>()
      1 cluster_data_directory = "/home/ubuntu/bitcoin"
----> 2 chain = blocksci.Blockchain(cluster_data_directory)

~/BlockSci/Notebooks/blocksci/__init__.py in new_init(self, loc)
    130 old_init = Blockchain.__init__
    131 def new_init(self, loc):
--> 132     old_init(self, loc)
    133     self.block_times = None
    134     self.cpp = CPP(self)

RuntimeError: failed opening file: Too many open files: iostream error
```

Seems a file is not being closed.

**Comment by hkalodner** [comment]
Hi @paulperry, there were some issues with that AMI. I released a new version of the AMI yesterday which I spawned on a r4.2 instance and verified works properly. Can try the new image and see if its working for you?

**Comment by paulperry** [comment]
@hkalodner : I launched a new AMI March 21, 2018 at 5:03:20 AM UTC-4 and I just got another `too many open files` error. Did I pick up the older AMI?

**Comment by hkalodner** [comment]
Hmm bizarre. The AMI should be ami-4405d039 which was the one linked from the readme when I last posted. What type of instance are you launching? I'm trying to recreate the problem, but I haven't been able to so far.

**Comment by paulperry** [comment]
Yes, the AMI is BlockSci v0.4.5 Release (ami-4405d039).  It takes a couple of days for the issue to come up.

**Comment by hkalodner** [comment]
Interesting, so it doesn't appear immediately? This probably has something to do with the RocksDB database we're using to store various pieces of data in the backend. This sounds less like a problem with BlockSci directly and more a problem with the Ubuntu configuration on the AMI. I'll try releasing a new copy of the AMI with the system file limit raised.

**Comment by hkalodner** [comment]
I just updated the [AMI](https://console.aws.amazon.com/ec2/home?region=us-east-1#launchAmi=ami-0650807b) again. The only change is that I increased the system file limit which will fix the problem assuming I'm correct that this error is just the result of using multiple RocksDB databases which each require opening a substantial number of files rather than the result of some sort of leak or bug within BlockSci. It should at least eliminate the `Too many open files` error.
